src/domain/enrolment/editButtonPanel/EditButtonPanel.tsx

```diff
@@ -0,0 +1,135 @@
+import classNames from 'classnames';
+import { IconArrowLeft, IconMenuDots } from 'hds-react';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useSelector } from 'react-redux';
+import { useHistory, useLocation } from 'react-router';
+import { toast } from 'react-toastify';
+
+import Button from '../../../common/components/button/Button';
+import MenuDropdown from '../../../common/components/menuDropdown/MenuDropdown';
+import { MenuItemOptionProps } from '../../../common/components/menuDropdown/MenuItem';
+import { ROUTES } from '../../../constants';
+import { Enrolment, Registration } from '../../../generated/graphql';
+import useIsMobile from '../../../hooks/useIsMobile';
+import useLocale from '../../../hooks/useLocale';
+import extractLatestReturnPath from '../../../utils/extractLatestReturnPath';
+import Container from '../../app/layout/Container';
+import { authenticatedSelector } from '../../auth/selectors';
+import { ENROLMENT_EDIT_ACTIONS } from '../../enrolments/constants';
+import { EnrolmentsLocationState } from '../../enrolments/types';
+import { getEditButtonProps } from '../../enrolments/utils';
+import FormContainer from '../formContainer/FormContainer';
+import styles from './editButtonPanel.module.scss';
+
+export interface EditButtonPanelProps {
+  enrolment: Enrolment;
+  onSave: () => void;
+  registration: Registration;
+}
+
+const EditButtonPanel: React.FC<EditButtonPanelProps> = ({
+  enrolment,
+  onSave,
+  registration,
+}) => {
+  const { t } = useTranslation();
+  const authenticated = useSelector(authenticatedSelector);
+  const locale = useLocale();
+  const isMobile = useIsMobile();
+  const history = useHistory<EnrolmentsLocationState>();
+  const { search } = useLocation();
+
+  const goBack = () => {
+    const { returnPath, remainingQueryString } = extractLatestReturnPath(
+      search,
+      ROUTES.REGISTRATION_ENROLMENTS.replace(
+        ':registrationId',
+        registration.id as string
+      )
+    );
+
+    history.push({
+      pathname: `/${locale}${returnPath}`,
+      search: remainingQueryString,
+      state: { enrolmentId: enrolment.id },
+    });
+  };
+
+  const getActionItemProps = ({
+    action,
+    onClick,
+  }: {
+    action: ENROLMENT_EDIT_ACTIONS;
+    onClick: () => void;
+  }): MenuItemOptionProps | null => {
+    return getEditButtonProps({
+      action,
+      authenticated,
+      enrolment,
+      onClick,
+      t,
+    });
+  };
+
+  const actionItems: MenuItemOptionProps[] = [
+    getActionItemProps({
+      action: ENROLMENT_EDIT_ACTIONS.SEND_MESSAGE,
+      onClick: () => toast.error('TODO: Send message to attendee'),
+    }),
+    getActionItemProps({
+      action: ENROLMENT_EDIT_ACTIONS.CANCEL,
+      onClick: () => toast.error('TODO: Cancel enrolment'),
+    }),
+  ].filter((i) => i) as MenuItemOptionProps[];
```

Comment:
Typed boolean helper could be used here to avoid type assertion:

```
export type FalsyType = false | null | undefined | '' | 0;

export const TypedBoolean = <ValueType>(
  value: ValueType
): value is Exclude<ValueType, FalsyType> => {
  return Boolean(value);
};
```

Can be used like this:

`].filter(TypedBoolean)`

Not sure about the best naming but in tapahtumat helsinki project we have: 

```
export const skipFalsyType = <ValueType>(
  value: ValueType
): value is Exclude<ValueType, FalsyType> => {
  return Boolean(value);
};
```

There is also some other places in the codebase where this could be used